pkg/pillar/scripts/dhcpcd.sh

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+set -e
+
+PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin
+export PATH
+
+TASK="$2"
+VIF_TASK=/run/tasks/vifs/"$TASK"
+
+case $1 in
+down)
+  unshare --mount --root="$VIF_TASK" dhcpcd --exit || :
+  umount "$VIF_TASK"/etc/dhcpcd.conf || :
+  umount "$VIF_TASK"/lib || :
+  umount "$VIF_TASK"/usr/lib || :
+  umount "$VIF_TASK"/bin || :
+  umount "$VIF_TASK"/usr/bin || :
+  umount "$VIF_TASK"/sbin || :
+  rm -rf "$VIF_TASK"
+  ;;
+up)
+  VIF_NS=$(jq '.pid')
+  mkdir -p "$VIF_TASK"
+
+  # mount files and directories required to run dhcpcd
+
+  mkdir -p "$VIF_TASK"/etc
+  touch "$VIF_TASK"/etc/dhcpcd.conf
+  mount --bind /etc/dhcpcd.conf "$VIF_TASK"/etc/dhcpcd.conf
+
+  mkdir -p "$VIF_TASK"/lib
+  mount --bind /lib "$VIF_TASK"/lib
+
+  mkdir -p "$VIF_TASK"/usr/lib
+  mount --bind /usr/lib "$VIF_TASK"/usr/lib
+
+  mkdir -p "$VIF_TASK"/bin
+  mount --bind /bin "$VIF_TASK"/bin
+
+  mkdir -p "$VIF_TASK"/sbin
+  mount --bind /sbin "$VIF_TASK"/sbin
+
+  mkdir -p "$VIF_TASK"/usr/bin
+  mount --bind /usr/bin "$VIF_TASK"/usr/bin
+
+  nsenter --target "$VIF_NS" --net unshare --uts --mount --root="$VIF_TASK" dhcpcd
```

Comment:
Old code passed $VIF_CTR as the first argument to dhcpcd. Don't we still need that?

Reply:
Now I start dhcpcd in manager mode, without defining of interfaces. We run it in network namespace, so it will see only interfaces associated with container. In case of manager mode we should terminate dhcpcd and I implement it few lines above with `--exit` option.

Reply:
But if the container has multiple interfaces, don't we want to run it explicitly for each interfaces? The manager AFAIU sits there waiting for different requests from dhcpcd invocations doesn't it?

Reply:
In [man](https://man.archlinux.org/man/core/dhcpcd/dhcpcd.8.en#Multiple_interfaces) I can see `If a list of interfaces are given on the command line, then dhcpcd only works with those interfaces, otherwise dhcpcd discovers available Ethernet interfaces that can be configured.`.

Reply:
Do any of the existing tests run containers with multiple interfaces?

Reply:
Seems only for containers-in-VM. Will add

Reply:
Included into https://github.com/lf-edge/eve/pull/2826

Reply:
With #2826 merged can you rebase so we can run with the added test? Or did you already do that manually?

Reply:
Sure, rebased.
Yes, I tested this manually.